### DISCUSSION TITLE
Need help with mocking http request within asyncClient using responses

### DISCUSSION THREAD

**Discussion by rathourarv** [encode/httpx#2591] [Q&A] [answered]
Hi All,

I have a method that makes an HTTP call to another service and I want to mock the API call using responses(https://github.com/getsentry/responses) on my test. The test tries to connect to the service instead of using a mock while running. 

Am I missing something here? how should I test this?

```
async def patch_account(
    *, url: str, data: dict[str, Any]
) -> dict[str, Any]:
    with httpx.AsyncClient() as client:
        response = await client.patch(
            url=url,
            data=json.dumps(data)
        )
        response.raise_for_status()
        response = response.json()
        return response
```

and test

```
async def test_patch_account():
    @responses.activate
    async def run():
        responses.patch(
            url="http://localhost:4500/account/1",
        )
        response = await patch_account(
            url="http://localhost:4500/account/1",
            data={"current_balance": 10}
        )
        assert response.status_code == 200
    await run()
```

Thanks!

**Comment by chbndrhnns** [comment, marked as answer]
The description of the `responses` package reads

> A utility for mocking out the Python Requests library.

Are you sure this works with httpx?

**Reply by lovelydinosaur** [reply]
You might want to try https://www.python-httpx.org/third_party_packages/#respx

**Reply by rathourarv** [reply]
It wasn't working with responses. I started using https://lundberg.github.io/respx/ for it. Thanks, @tomchristie!
